tests/step_implementers/push_artifacts/test_maven_push_artifacts.py

```diff
@@ -1,24 +1,381 @@
-        config = {
-            'tssc-config': {    
-                'push-artifacts': {
-                    'implementer': 'Maven',
-                    'config': {}
+class TestStepImplementerPushArtifact(unittest.TestCase):
+
+    # ------------ SIMPLE tests that test the config required items
+    @patch('sh.mvn', create=True)
+    def test_push_artifact_with_repository_url_missing_from_config(self, mvn_mock):
+        with TempDirectory() as temp_dir:
+            config = {
+                'tssc-config': {
+                    'push-artifacts': {
+                        'implementer': 'Maven',
+                        'config': {
+                            'user': 'frankie',
+                            'password': 'goestohollywood',
+                        }
+                    }
                 }
             }
-        }
-        expected_step_results = {'tssc-results': {'push-artifacts': {}}}
+            passed = False
+            expected_step_results = {}
+            try:
+              run_step_test_with_result_validation(temp_dir, 'push-artifacts', config, expected_step_results)
+            except ValueError as err:
+                if err.__str__() == 'Key (url) must have none empty value in the step configuration':
+                    passed = True
+        assert passed
+
+    @patch('sh.mvn', create=True)
+    def test_push_artifact_with_user_missing_from_config(self, mvn_mock):
+        with TempDirectory() as temp_dir:
+            config = {
+                'tssc-config': {
+                    'push-artifacts': {
+                        'implementer': 'Maven',
+                        'config': {
+                            'password': 'goestohollywood',
+                            'url': 'http://artifactory.apps.tssc.rht-set.com/artifactory/tssc/',
+                        }
+                     }
+                 }
+             }
+            passed = False
+            expected_step_results = {}
+            try:
+              run_step_test_with_result_validation(temp_dir, 'push-artifacts', config, expected_step_results)
+            except ValueError as err:
+                if err.__str__() == 'Key (user) must have none empty value in the step configuration':
+                    passed = True
+        assert passed
+
+    @patch('sh.mvn', create=True)
+    def test_push_artifact_with_password_missing_from_config(self, mvn_mock):
+        with TempDirectory() as temp_dir:
+            config = {
+                'tssc-config': {
+                    'push-artifacts': {
+                        'implementer': 'Maven',
+                        'config': {
+                            'user': 'frankie',
+                            'url': 'http://artifactory.apps.tssc.rht-set.com/artifactory/tssc/',
+                        }
+                    }
+                }
+            }
+            passed = False
+            expected_step_results = {}
+            try:
```

Comment:
@ppremru fix this

Reply:
Updated to use the assertRaisesRegex() pattern